### PR TITLE
ICU-22757 Remove allow list of known contractions with precomposed form from ICU4X mode of genuca

### PR DESCRIPTION
**Pull request by hsivonen**
This assumes that future cases will work OK, since the addition that was seen in Unicode 16 alpha was OK.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22757
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/data/unidata/generate.sh [hunk 1]
```diff
@@ -49,6 +49,5 @@ bazelisk run //tools/unicode/c/genprops $ICU_SRC/icu4c
 bazelisk run //tools/unicode/c/genuca -- --hanOrder implicit $ICU_SRC/icu4c
 bazelisk run //tools/unicode/c/genuca -- --hanOrder radical-stroke $ICU_SRC/icu4c
 # Also generate the ICU4X versions
-# TODO: Currently fails with early Unicode 16.0 FractionalUCA.txt.
-# bazelisk run //tools/unicode/c/genuca -- --icu4x --hanOrder implicit $ICU_SRC/icu4c
-# bazelisk run //tools/unicode/c/genuca -- --icu4x --hanOrder radical-stroke $ICU_SRC/icu4c
+bazelisk run //tools/unicode/c/genuca -- --icu4x --hanOrder implicit $ICU_SRC/icu4c
```
Comment (markusicu) [inline icu4c/source/data/unidata/generate.sh:52]: Please also restore the "rm" lines above to their normal state (before my PR #2930).

File: icu4c/source/i18n/collationdatabuilder.cpp [hunk 2]
```diff
@@ -586,27 +586,17 @@ CollationDataBuilder::addCE32(const UnicodeString &prefix, const UnicodeString &
...
-            // Unknown case worth investigating if ever found.
-            errorCode = U_UNSUPPORTED_ERROR;
+
+            // As of Unicode 16 alpha, the cases that come here are:
+            //
+            // 1. The second character is a special decomposing Tibetan vowel
+            //    sign. These are OK to ignored in the precomposed form, since
+            //    the decomposed form is added also.
+            // 2. Likewise for KIRAT RAI VOWEL SIGN AA followed by
+            //    KIRAT RAI VOWEL SIGN AI.
```
Comment (markusicu) [inline icu4c/source/i18n/collationdatabuilder.cpp:596]:
```suggestion
            // 2. Likewise for KIRAT RAI VOWEL SIGN AA followed by KIRAT RAI VOWEL SIGN AI
            //    and other such cases.
            //    For details see the normalization section of
            //    https://www.unicode.org/review/pri497/pri497-background.html
```

**Review by markusicu — COMMENTED** [review]
mostly lgtm
I believe that the test failure is due to this PR not also updating the generated binary data files.

File: icu4c/source/data/unidata/generate.sh (hunk 1 above)
Comment (hsivonen) [reply icu4c/source/data/unidata/generate.sh:51]: Uncommented on `rm` line.

File: icu4c/source/data/unidata/generate.sh [hunk 3]
```diff
@@ -23,8 +23,7 @@ rm $ICU_SRC/icu4c/source/common/propname_data.h
 rm $ICU_SRC/icu4c/source/common/*_props_data.h
 rm $ICU4C_DATA_IN/*.icu
 rm $ICU4C_DATA_IN/*.nrm
-# TODO: Back to deleting coll/*.icu once ICU4X data generation is fixed.
-# rm $ICU4C_DATA_IN/coll/*.icu
+rm $ICU4C_DATA_IN/coll/*.icu
```
Comment (markusicu) [inline icu4c/source/data/unidata/generate.sh:26]: Please also remove the following two lines, redundant with this one, to bring this part back to how it was before.

**Review by markusicu — COMMENTED** [review]
one cleanup left

I don't understand how the [build failure](https://dev.azure.com/ms/icu/_build/results?buildId=571246&view=logs&jobId=d69c23bb-4a3f-572d-3cb6-52938b11d573&j=d69c23bb-4a3f-572d-3cb6-52938b11d573&t=c420e7af-7ce3-5423-9ebf-70909c5f817e) relates to these changes. I will try to restart that build.

File: icu4c/source/data/unidata/generate.sh (hunk 3 above)
Comment (hsivonen) [reply icu4c/source/data/unidata/generate.sh:26]: Fixed.

**Review by markusicu — APPROVED** [review]

**Pull request merged**